src/ts/components/quickseekbutton.ts

```diff
@@ -0,0 +1,131 @@
+import { Button, ButtonConfig } from './button';
+import { i18n } from '../localization/i18n';
+import { PlayerAPI } from 'bitmovin-player';
+import { UIInstanceManager } from '../uimanager';
+import { PlayerUtils } from '../playerutils';
+
+export enum QuickSeekDirection {
+  Forward = 'forward',
+  Rewind = 'rewind',
+}
+
+export interface QuickSeekButtonConfig extends ButtonConfig {
+  /**
+   * Specify how many seconds the player should seek forward/backwards in the stream.
+   * Default is 10.
+   */
+  seekSeconds?: number;
+
+  /**
+   * Specify whether the button should fast-forward or rewind.
+   * Default is rewind.
+   */
+  direction?: QuickSeekDirection;
+}
+
+export class QuickSeekButton extends Button<QuickSeekButtonConfig> {
+  constructor(config: QuickSeekButtonConfig = {}) {
+    super(config);
+
+    this.config = this.mergeConfig(
+      config,
+      {
+        seekSeconds: 10,
+        direction: QuickSeekDirection.Rewind,
+        cssClass: 'ui-quickseekbutton',
+      },
+      this.config,
+    );
+
+    this.config.text =
+      this.config.text ||
+      i18n.getLocalizer(`quickseek.${this.config.direction}`);
+    this.config.ariaLabel =
+      this.config.ariaLabel ||
+      i18n.getLocalizer(`quickseek.${this.config.direction}`);
+
+    this.getDomElement().data(
+      this.prefixCss('seek-direction'),
+      this.config.direction,
+    );
+  }
+
+  configure(player: PlayerAPI, uimanager: UIInstanceManager): void {
+    super.configure(player, uimanager);
+
+    let isLive: boolean;
+    let hasTimeShift: boolean;
+
+    const switchVisibility = (isLive: boolean, hasTimeShift: boolean) => {
+      if (isLive && !hasTimeShift) {
+        this.hide();
+      } else {
+        this.show();
+      }
+    };
+
+    const timeShiftDetector = new PlayerUtils.TimeShiftAvailabilityDetector(
+      player,
+    );
+    timeShiftDetector.onTimeShiftAvailabilityChanged.subscribe(
+      (sender, args: PlayerUtils.TimeShiftAvailabilityChangedArgs) => {
+        hasTimeShift = args.timeShiftAvailable;
+        switchVisibility(isLive, hasTimeShift);
+      },
+    );
+
+    let liveStreamDetector = new PlayerUtils.LiveStreamDetector(
+      player,
+      uimanager,
+    );
+    liveStreamDetector.onLiveChanged.subscribe(
+      (sender, args: PlayerUtils.LiveStreamDetectorEventArgs) => {
+        isLive = args.live;
+        switchVisibility(isLive, hasTimeShift);
+      },
+    );
+
+    // Initial detection
+    timeShiftDetector.detect();
+    liveStreamDetector.detect();
+
+    this.onClick.subscribe(() => {
```

Comment:
Probably out of scope, but it would be great if these would also trigger the seeks:
- Pressing Left/Right arrow key
- Double-tapping the left/right half of the video on mobile

Maybe for a follow-up :)

Reply:
I agree, I think this is out of scope.


> - Pressing Left/Right arrow key

I agree, but then this might interfere with the functionality we already have in https://github.com/bitmovin/bitmovin-player-ui/blob/6e05cbdfc2d67ea3e11338accc086671b1c53054/src/ts/components/seekbarcontroller.ts#L83-L92


> - Double-tapping the left/right half of the video on mobile

I think this and/or a "HugeQuickSeekButton" for displaying within the video would be separate components (similar to `PlaybackToggleButton` and `HugePlaybackToggleButton`.

Reply:
Yup, makes sense. Just wanted to note it down :D